src/script/standard.cpp

```diff
@@ -51,13 +51,10 @@ bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, vector<vector<unsi
 
         // Sender provides N pubkeys, receivers provides M signatures
         mTemplates.insert(make_pair(TX_MULTISIG, CScript() << OP_SMALLINTEGER << OP_PUBKEYS << OP_SMALLINTEGER << OP_CHECKMULTISIG));
-
-        // Empty, provably prunable, data-carrying output
-        if (GetBoolArg("-datacarrier", true))
```

Comment:
I agree with getting rid of the -datacarrier=0 option, it's just a redundant way to say -datacarriersize=0. But if you're removing the option, you should also get rid of https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L400 and raise a deprecated error when -datacarrier is used before https://github.com/bitcoin/bitcoin/blob/master/src/init.cpp#L872